starknet_py/net/client_models.py

```diff
@@ -62,16 +62,23 @@ class Event:
 
 
 @dataclass
-class EmittedEvent(Event):
-    """
-    Dataclass representing an event emitted by transaction.
-    """
-
+class _EmittedEventBase(Event):
     transaction_hash: int
+
+
+@dataclass
+class _EmittedEventDefaultBase(Event):
     block_hash: Optional[int] = None
     block_number: Optional[int] = None
 
 
+@dataclass
+class EmittedEvent(_EmittedEventDefaultBase, _EmittedEventBase):
```

Comment:
I think technically this is a breaking change if someone was creating this struct for whatever reasons using positional arguments.

But since it's a super weird use-case, I don't think we need to mention that anywhere.